### DISCUSSION TITLE
Python Analyzer for lucene

### DISCUSSION THREAD

**Discussion by mustafa0x** [coady/lupyne#27] [General]
I'm using classical solr/lucene. I'd like to write a python analyzer (part-of-speech NLP library). I couldn't find a good way to do this using solr/lucene. I came across this file which makes me think lupyne might be a solution:

https://github.com/coady/lupyne/blob/main/lupyne/engine/analyzers.py 

Or should I instead process the text before inserting into lucene? Eg replace `<word>` with `<pos>|<word>`, then tell lucene to split at `|`?

**Comment by coady** [comment]
Lucene supports custom analyzers, and lupyne makes writing one on easier. I'm not as familiar with Solr, but it appears to supports custom analyzers through [plugins](https://solr.apache.org/guide/solr/latest/configuration-guide/solr-plugins.html).

**Reply by mustafa0x** [reply]
Thanks for the reply!

Yes, solr does support custom analyzers, but the problem is that they have to be in Java.

https://solr.apache.org/guide/solr/latest/indexing-guide/analyzers.html

> The simplest way to configure an analyzer is with a single <analyzer> element whose class attribute is a fully qualified Java class name.

However the library I plan to use is in Python, hence my questions.

Thanks again!
